client/src/pages/Resource/index.js

```diff
@@ -0,0 +1,78 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+
+import { ResourceService, useService } from "../../services";
+
+import "./Resource.scss";
+
+export default function Resource() {
+	const [resource, setResource] = useState(undefined);
+	const resourceService = useService(ResourceService);
+	const { id } = useParams();
+
+	useEffect(() => {
+		resourceService.getOne(id).then(setResource);
+	}, [id, resourceService]);
+
+	if (!resource) {
+		return null;
+	}
+
+	return (
+		<>
+			<h2>
+				Resource &gt; <span>{resource.title}</span>
+			</h2>
+			<section>
+				<table aria-label="resource details" className="resource-details">
+					<tbody>
+						<tr>
+							<th>Title</th>
```

Comment:
It's another option, I don't know if it's _better_ or not. Explicit property access on `resource` means it's very clear where those things are coming from, rather than adding another 7 names into the scope of the function to remember, but it does mean a bit more typing.

Note that `|| {}` should be `?? {}` in modern JS, and given that we wouldn't reach this line if it was still null it's not really needed at all.